src/ConfigParser.cc

```diff
@@ -676,3 +676,12 @@ ConfigParser::CfgFile::~CfgFile()
         fclose(wordFile);
 }
 
+Configuration::TokensIterator
+Configuration::Tokens::begin() const
+{
+    const auto first = TokensIterator(parser_, method_);
+    if (!emptyAllowed_ && first == end())
+        throw MissingTokenException(ToSBuf("missing required argument: ", description_), Here());
```

Comment:
If possible, let's generalize/simplify the error message by not assuming that Tokens are "required arguments":

```suggestion
        throw MissingTokenException(ToSBuf("missing ", description_), Here());
```